### PR TITLE
Add CBZ Grow Staking

### PR DESCRIPTION
**Pull request by 73kn1k**
Nice repo. Hope all ok. Tnks!

### CODE REVIEW

**Comment by 73kn1k** [comment]
@raymondfeng can you please have a look.

Thank you

File: src/adapters/cbzgrow.adapter.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      this.provider,
+    );
+    if (name === 'CoenerBoyz') {
+      return contract.stakedCBZs(owner);
+    } else if (name === 'CBZ Seeds') {
+      return contract.stakedSeeds(owner);
+    } else {
+      const tmpval = [];
+      tmpval[0] = BigNumber.from(0);
+      return Promise.resolve(tmpval);
```
Comment (raymondfeng) [inline src/adapters/cbzgrow.adapter.ts:67]: Let make the function `async`:
```ts
async getStakedTokenIds(owner: string, name?: string) {
const contract = Cbzgrow__factory.connect(
      this.contractAddress,
      this.provider,
    );
    if (name === 'CoenerBoyz') {
      return contract.stakedCBZs(owner);
    } else if (name === 'CBZ Seeds') {
      return contract.stakedSeeds(owner);
    } else {
      return [];
    }
}

File: src/adapters/cbzgrow.adapter.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return resp.cbzStaked;
+      } else if (assetName === 'CBZ Seeds') {
+        return resp.seedStaked;
+      } else {
+        return Promise.resolve(BigNumber.from(0));
+      }
+    }
+    return run();
+  }
+}
```
Comment (raymondfeng) [inline src/adapters/cbzgrow.adapter.ts:93]:
```ts
async getStakedTokenBalance(owner: string, assetName?: string) {
    const contract = Cbzgrow__factory.connect(
      this.contractAddress,
      this.provider,
    );
      const resp = await contract.stakers(owner);
      if (assetName === 'CoenerBoyz') {
        return resp.cbzStaked;
      } else if (assetName === 'CBZ Seeds') {
        return resp.seedStaked;
      } else {
        return BigNumber.from(0);
      }
  }
```

File: src/adapters/cbzgrow.adapter.ts (hunk 1 above)
Comment (73kn1k) [reply src/adapters/cbzgrow.adapter.ts:67]: only that function or getStakedTokenBalance to?

Comment (73kn1k) [reply src/adapters/cbzgrow.adapter.ts:67]: done

File: src/adapters/cbzgrow.adapter.ts (hunk 2 above)
Comment (73kn1k) [reply src/adapters/cbzgrow.adapter.ts:93]: done

**Review by 73kn1k — COMMENTED** [review]
Thanks for teh review.
All changes added.

File: src/adapters/cbzgrow.adapter.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      this.provider,
+    );
+    async function run() {
+      const resp = await contract.stakers(owner);
+      if (assetName === 'CoenerBoyz') {
+        return resp.cbzStaked;
+      } else if (assetName === 'CBZ Seeds') {
+        return resp.seedStaked;
+      } else {
+        return Promise.resolve(BigNumber.from(0));
```
Comment (raymondfeng) [inline src/adapters/cbzgrow.adapter.ts:88]: You can simply `return BigNumber.from(0)`.

File: src/adapters/cbzgrow.adapter.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      }
+    }
+    return run();
+  }
+}
```
Comment (raymondfeng) [inline src/adapters/cbzgrow.adapter.ts:93]: We don't have to wrap the logic into `run()`. Let's flatten it as suggested.

Comment (73kn1k) [reply src/adapters/cbzgrow.adapter.ts:93]: ok will correct.

File: src/adapters/cbzgrow.adapter.ts [hunk 5]
```diff
@@ -78,16 +78,13 @@ export class CbzGrowStakingContractAdapter extends BaseStakingContractAdapter {
...
-      }
+    const resp = await contract.stakers(owner);
+    if (assetName === 'CoenerBoyz') {
+      return resp.cbzStaked;
+    } else if (assetName === 'CBZ Seeds') {
+      return resp.seedStaked;
+    } else {
+      return BigNumber.from(0);
     }
-    return run();
```
Comment (73kn1k) [inline src/adapters/cbzgrow.adapter.ts]: @raymondfeng here made the modification.

**Review by 73kn1k — COMMENTED** [review]
hi @raymondfeng tnks for the review.
here i outwrapped the function and replace the return value to BigNumer done.

File: src/adapters/cbzgrow.adapter.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      this.provider,
+    );
+    if (name === 'CoenerBoyz') {
+      return contract.stakedCBZs(owner);
+    } else if (name === 'CBZ Seeds') {
+      return contract.stakedSeeds(owner);
+    } else {
+      const tmpval = [];
+      tmpval[0] = BigNumber.from(0);
+      return Promise.resolve(tmpval);
```
Comment (raymondfeng) [inline src/adapters/cbzgrow.adapter.ts:67]: Why don't you return `[]`instead?

File: src/adapters/cbzgrow.adapter.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    const contract = Cbzgrow__factory.connect(
+      this.contractAddress,
+      this.provider,
+    );
+    if (name === 'CoenerBoyz') {
+      return contract.stakedCBZs(owner);
+    } else if (name === 'CBZ Seeds') {
+      return contract.stakedSeeds(owner);
+    } else {
+      return [];
```
Comment (73kn1k) [inline src/adapters/cbzgrow.adapter.ts:65]: here done as sugested

**Comment by 73kn1k** [comment]
Hi @raymondfeng, i recived a mail. and see it has a problem. And my cuestion is. Is this only for polygon assets or ethereum to... cuz i think that is the problem. CBZ grow is in ethereum.

**Pull request merged**